### PR TITLE
Fix for acceptance tests

### PR DESCRIPTION
**Pull request by absolutelightning**
Changes proposed in this PR:
- Fixes acceptance test failures introduced in - https://github.com/hashicorp/consul-k8s/pull/2841/files

How I've tested this PR:
Ran the test on local.

How I expect reviewers to test this PR:

Checklist:
- [x] Tests added
- [] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by absolutelightning** [comment]
I am updating the bats test now. Please keep go ahead with review.

**Review by zalimeni — DISMISSED** [review]

**Comment by zalimeni** [comment]
Reran failed test since it looks like a flake, given the prior run succeeded and only a test name was changed

**Comment by zalimeni** [comment]
@absolutelightning it looks like the original change was backported to `1.0.x` and `1.1.x`. We typically don't backport features, so it might make sense to revert those changes. If not, I'd expect we'll need this fix backported as well.

File: charts/consul/test/unit/terminating-gateways-deployment.bats
Comment (zalimeni) [inline charts/consul/test/unit/terminating-gateways-deployment.bats:1239]: I think this test is currently failing: https://github.com/hashicorp/consul-k8s-workflows/actions/runs/6149328162/job/16692536295#step:6:2216

Comment (absolutelightning) [reply charts/consul/test/unit/terminating-gateways-deployment.bats:1239]: This was running fine on my local. Had to make some syntactical changes to make it pass on CI.

**Review by Ganeshrockz — APPROVED** [review]
Approving. Do wait for the CI to be green before merging this.

**Comment by absolutelightning** [comment]
Acceptance CNI is - Partition test is only failing but it is not related to my change. Hence merging the PR since CI is 99% green.

**Pull request merged**
